macOS clang warnings

```
gcc -I. -c -DBINDIR=\"/usr/local/bin\" -DSYSDIR=\"/usr/local/etc\"  -g -O2 option.c
screen.c:741:19: warning: passing 'const char *' to parameter of type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        return (tgetflag(capname));
                         ^~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/termcap.h:66:58: note: passing argument to parameter here
extern NCURSES_EXPORT(int) tgetflag (NCURSES_CONST char *);
                                                         ^
screen.c:752:18: warning: passing 'const char *' to parameter of type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        return (tgetnum(capname));
                        ^~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/termcap.h:67:57: note: passing argument to parameter here
extern NCURSES_EXPORT(int) tgetnum (NCURSES_CONST char *);
                                                        ^
screen.c:763:18: warning: passing 'const char *' to parameter of type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        return (tgetstr(capname, pp));
                        ^~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/termcap.h:63:60: note: passing argument to parameter here
extern NCURSES_EXPORT(char *) tgetstr (NCURSES_CONST char *, char **);
                                                           ^

--- a/screen.c
+++ b/screen.c
@@ -103,6 +103,9 @@ extern int fd0;
 #ifdef _OSK
 #include <signal.h>
 #endif
+#ifdef __APPLE__
+#pragma clang diagnostic ignored "-Wincompatible-pointer-types-discards-qualifiers"
+#endif
 #if OS2
 #include <sys/signal.h>
 #include "pckeys.h"

